pkg/network/setup/netstat_test.go

```diff
@@ -841,6 +843,48 @@ var _ = Describe("netstat", func() {
 				},
 			}))
 		})
+
+		It("does not report all guest interfaces when their count exceeds the limit", func() {
+			var guestAgentIfaces []api.InterfaceStatus
+			const maxGuestIfacesReported = 10
+			for i := 1; i < (maxGuestIfacesReported + 10); i++ {
+				ifaceMac := fmt.Sprintf("00:00:00:00:00:%02X", i)
+				ifaceIpv4 := fmt.Sprintf("10.10.10.%d", i)
+				ifaceIpv6 := fmt.Sprintf("ffff::%x", i)
+				guestAgentIfaces = append(guestAgentIfaces,
+					newDomainStatusIface([]string{ifaceIpv4, ifaceIpv6}, ifaceMac, fmt.Sprintf("eth%d", i)),
+				)
+			}
+			guestAgentIfaces = append(guestAgentIfaces,
+				newDomainStatusIface([]string{primaryGaIPv4, primaryGaIPv6}, primaryMAC, primaryIfaceName),
+				newDomainStatusIface([]string{secondaryGaIPv4, secondaryGaIPv6}, secondaryMAC, secondaryIfaceName),
+			)
+			setup.addGuestAgentInterfaces(guestAgentIfaces...)
+			Expect(setup.NetStat.UpdateStatus(setup.Vmi, setup.Domain)).To(Succeed())
+			Expect(setup.Vmi.Status.Interfaces).To(HaveLen(maxGuestIfacesReported + 2))
+			Expect(setup.Vmi.Status.Interfaces).To(ContainElements([]v1.VirtualMachineInstanceNetworkInterface{
```

Comment:
Can we assert equality for the whole slice against an expected slice by using a helper that will generate the guest internal slice?